### PR TITLE
Multisampling and cacheAsBitmap

### PR DESCRIPTION
**Pull request by dev7355608**
##### Description of change

Added `cacheAsBitmapMultisample`. Default behavior (if `cacheAsBitmapMultisample` is undefined): it renders with `HIGH` samples if the renderer has antialias enabled; if antialias is not enabled, it renders without multisampling.

##### Pre-Merge Checklist

- [ ] Tests and/or benchmarks are included
- [x] Documentation is changed or added
- [x] Lint process passed (`npm run lint`)
- [x] Tests passed (`npm run test`)

### CODE REVIEW

**Review by bigtimebuddy — APPROVED** [review]
I think this is a good idea. Makes cacheAsBitmap more powerful.

**Review by ivanpopelyshev — CHANGES_REQUESTED** [review]

File: packages/mixin-cache-as-bitmap/src/index.ts [hunk 1]
```diff
@@ -4,13 +4,15 @@ import { Container, DisplayObject, IDestroyOptions } from '@pixi/display';
...
 import { settings } from '@pixi/settings';
+import { MSAA_QUALITY } from '@pixi/constants';
 import type { CanvasRenderer } from '@pixi/canvas-renderer';
 
 const _tempMatrix = new Matrix();
 
 DisplayObject.prototype._cacheAsBitmap = false;
 DisplayObject.prototype._cacheData = null;
 DisplayObject.prototype._cacheAsBitmapResolution = null;
+DisplayObject.prototype._cacheAsBitmapMultisample = undefined;
```
[the remark this answers is not in the document]
Comment (dev7355608) [reply packages/mixin-cache-as-bitmap/src/index.ts:15]: I chose the same default behavior as `cacheAsBitmapResolution`: if not defined explicitly, use what the renderer is using. I think this makes the most sense for cacheAsBitmap and that resolution and multisample have the same default behavior. If we render the object once (or every infrequently), we might as well render it in good quality, I think.

File: packages/mixin-cache-as-bitmap/src/index.ts [hunk 2]
```diff
@@ -247,6 +282,15 @@ DisplayObject.prototype._initCachedDisplayObject = function _initCachedDisplayOb
         resolution: this.cacheAsBitmapResolution || renderer.resolution,
     });
 
+    if (this.cacheAsBitmapMultisample !== undefined)
+    {
+        renderTexture.framebuffer.multisample = this.cacheAsBitmapMultisample;
+    }
+    else if (renderer.gl.getContextAttributes().antialias)
+    {
+        renderTexture.framebuffer.multisample = MSAA_QUALITY.HIGH;
```
Comment (SukantPal) [inline packages/mixin-cache-as-bitmap/src/index.ts:291]: Multisampling was introduced as an experimental feature. I'd be careful in making it default behavior just yet.

**Review by SukantPal — APPROVED** [review]
Otherwise nice work

File: packages/mixin-cache-as-bitmap/src/index.ts (hunk 2 above)
Comment (bigtimebuddy) [reply packages/mixin-cache-as-bitmap/src/index.ts:291]: Agreed

File: packages/mixin-cache-as-bitmap/src/index.ts [hunk 3]
```diff
@@ -247,6 +282,15 @@ DisplayObject.prototype._initCachedDisplayObject = function _initCachedDisplayOb
...
+    }
```
Comment (bigtimebuddy) [inline packages/mixin-cache-as-bitmap/src/index.ts:292]: I this this is good, but it is a change in the default behavior and should be considered a breaking change. We can either remove it or provide a static property to opt-in to this behavior, then make it the default in v7. Thoughts @dev7355608?
 
```suggestion
```

Comment (dev7355608) [reply packages/mixin-cache-as-bitmap/src/index.ts:292]: I removed it. I think we don't really need an opt-in property. For example, anyone who wants to force multisampling everywhere could simply change the default like this.
```js
DisplayObject.prototype._cacheAsBitmapMultisample = PIXI.MSAA_QUALITY.HIGH;
```

**Pull request merged**